cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,243 @@
+in XML documents.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Envelope](#4-envelope)
+5. [XML Batch Format](#5-xml-batch-format)
+6. [Examples](#6-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how
+elements defined in the CloudEvents specification are to be represented using
+[Extensible Markup Language (XML)](xml-spec) documents.
+
+* The [Attributes](#2-attributes) section describes the representation and
+data type mappings for CloudEvents context attributes.
+
+* The [Data](#3-data) section defines the container for the data portion of a
+CloudEvent.
+
+* The [Envelope](#4-envelope) section defines an XML container for CloudEvents
+attributes and an associated media type.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+may be packaged into a single XML document.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This specification is deliberately verbose in favor of readability and
+deterministic processing.
+
+Preservation of attribute value type information is supported allowing
+custom extensions to be communicated without value type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to the XML schema types as follows :
+
+| CloudEvents   |  XML Schema Type| Notes |
+| :-------------| :---------------------------------------------------------- |
+| Boolean       | [xs:boolean][xml-primitives] | |
+| Integer       | [xs:int][xml-primitives] | |
+| String        | [xs:string][xml-primitives] | |
+| Binary        | [xs:base64Binary][xml-primitives] | |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each context attribute is represented as an XML element whose name exactly matches that of
+a required or optional CloudEvent attribute, extension attribute names `MUST` adhere and
+align with the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes `SHOULD` be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator `SHOULD` be interpreted with an implied type of `xs:string`
```

Comment:
For well-known attributes shouldn't they follow the type defined in the CE spec? As of today I think they're all 'strings' but if that changes it seems like this spec should allow for non-strings.

Reply:
The spec defines those 7 core types, all of which *may* be represented as string types .. but using the appropriate xsi:type makes everything explicit.

Reply:
Sorry,I think I misread things... or I can't remember what I was thinking. Ignore my comment.